### PR TITLE
added date validation for cohort

### PR DESCRIPTION
**Pull request by rmonica08**
1. Added date validation for program cohort
2. Updated create new program cohort test - to create the record from the listing page and not from the program page
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
- [ ] Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))
- [ ] Base axe-core JEST test included for any new LWC (No critical violations)
- [ ] CRUD/FLS is enforced in Apex
- [ ] Permission sets are updated to account for CRUD/FLS for new object metadata
- [ ] All modified classes are unit tested (Apex) at 100% coverage
- [ ] UX approval or UX not necessary
- [ ] PR contains draft release notes
- [ ] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [ ] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [ ] QE story level testing completed

### CODE REVIEW

File: robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot [hunk 1]
```diff
@@ -15,29 +15,73 @@ Setup Test Data
...
+    &{program} =                API Create Program  
+    Store Session Record        ${ns}Program__c         ${program}[Id]
+    Set suite variable          &{program}
 
 
 *** Test Cases ***
-
-Create Program Cohort from Program Object
+Create Program Cohort
+     [Documentation]                        This test creates a program cohort and verifies the details of the page
```
Comment (skristem) [inline robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot:26]: can u expand on what details are verified on the page

File: robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot [hunk 2]
```diff
@@ -15,29 +15,73 @@ Setup Test Data
...
+     [tags]                                 feature:Program Cohort
```
Comment (skristem) [inline robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot:27]: can u add WI tag here? also, use unstable tag for all new tests to start with and once u r sure they are stable, move to stable flow

File: robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot [hunk 3]
```diff
@@ -15,29 +15,73 @@ Setup Test Data
...
+     populate lookup field                  Program                 ${program}[Name]
+     populate field                         Description             New Program Cohort via Robot Test
+     select date from datepicker            Start Date              25
+     select date from datepicker            End Date                10
+     Click modal button                     Save
+     verify modal error                     Start Date must be before End Date
+
+Date validation
+    [Documentation]                         This test opens the new program dialog and enters a end date earlier than start date
+    ...                                     and verifies that an error message is displayed
```
Comment (skristem) [inline robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot:67]: can u modify this documentation to match what exactly being done like open program edit dates and save. create program coharts from related list so on

Comment (rmonica08) [reply robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot]: updated the documentation

File: robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot (hunk 2 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot]: added WI tags

File: robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot (hunk 1 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot]: updated the documentation and also fixed cases for keywords

File: robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot [hunk 4]
```diff
@@ -15,29 +15,75 @@ Setup Test Data
     ${ns} =                 Get PMM Namespace Prefix
     Set suite variable      ${ns}
     ${program_cohort} =     Generate Random String
-    Set suite variable      ${program_cohort}
-    &{program} =            API Create Program
-    Store Session Record    ${ns}Program__c         &{program}[Id]
-    Set suite variable      &{program}
+    Set suite variable      ${program_cohort}      
+    &{program} =                API Create Program  
+    Store Session Record        ${ns}Program__c         ${program}[Id]
```
Comment (skristem) [inline robot/pmm/tests/browser/ProgramCohort/Create_Program_Cohort.robot:20]: remove this line as its already taken care by api keyword

**Pull request merged**